skbio/sequence/_grammared_sequence.py

```diff
@@ -209,6 +211,21 @@ def _noncanonical_codes(cls):
             cls.__noncanonical_codes = np.asarray([ord(c) for c in noncanonical_chars])
         return cls.__noncanonical_codes
 
+    @classproperty
+    def _degenerate_hash(cls):
+        if cls.__degenerate_hash is None:
+            htab = np.zeros((256,), dtype=np.uint8)
```

Comment:
This can be the following to avoid creating an extra variable (`htab`):

```python
cls.__degenerate_hash = np.zeros((256,), dtype=np.uint8)
cls.__degenerate_hash[cls._degenerate_codes] = 1
```